tests/CMakeLists.txt

```diff
@@ -6,18 +6,36 @@ if(NOT BOOST_FOUND)
   return()
 endif()
 
-set(TEST_LIBRARIES ${Boost_UNIT_TEST_FRAMEWORK_LIBRARY} ZeroEQ ZeroEQHBP
+set(TEST_LIBRARIES ${Boost_UNIT_TEST_FRAMEWORK_LIBRARY} ZeroEQ
   ${CMAKE_THREAD_LIBS_INIT} ${LIBZMQ_LIBRARIES})
-flatbuffers_generate_c_headers(FBS_NEW_EVENT newEvent.fbs)
-set_source_files_properties(
-  newEvent.cpp PROPERTIES OBJECT_DEPENDS ${FBS_NEW_EVENT_OUTPUTS})
 
-flatbuffers_generate_c_headers(FBS_BINARY_EVENT binary.fbs)
-set_source_files_properties(
-  serialization.cpp PROPERTIES OBJECT_DEPENDS ${FBS_BINARY_EVENT_OUTPUTS})
+if(FLATBUFFERS_FOUND)
+  flatbuffers_generate_c_headers(FBS_NEW_EVENT newEvent.fbs)
+  set_source_files_properties(
+    newEvent.cpp PROPERTIES OBJECT_DEPENDS ${FBS_NEW_EVENT_OUTPUTS})
+  flatbuffers_generate_c_headers(FBS_EMPTY_EVENT emptyEvent.fbs)
+  set_source_files_properties(
+    pubSub.cpp PROPERTIES OBJECT_DEPENDS ${FBS_EMPTY_EVENT_OUTPUTS})
+  set_source_files_properties(
+    perf.cpp PROPERTIES OBJECT_DEPENDS ${FBS_EMPTY_EVENT_OUTPUTS})
+  flatbuffers_generate_c_headers(FBS_BINARY_EVENT binary.fbs)
+  list(APPEND TEST_LIBRARIES ZeroEQHBP)
+endif()
 
+set(EXCLUDE_FROM_TESTS)
 if(NOT HTTPXX_FOUND)
-  set(EXCLUDE_FROM_TESTS http/server.cpp)
+  list(APPEND EXCLUDE_FROM_TESTS http/server.cpp)
+endif()
+
+if(NOT FLATBUFFERS_FOUND)
+  list(APPEND EXCLUDE_FROM_TESTS connection/broker.cpp
+                                 hbp/serialization.cpp
+                                 newEvent.cpp
+                                 perf.cpp
+                                 publisher.cpp
+                                 pubSub.cpp
+                                 receiver.cpp
+                                 subscriber.cpp)
```

Comment:
Oh no... we should not loose these tests with FB. They should be run at least with a minimal test object (similar to Servus/tests/serializable.cpp) and optional extended tests using the FB objects